posthog/hogql/resolver.py

```diff
@@ -573,3 +581,18 @@ def lookup_cte_by_name(scopes: List[ast.SelectQueryType], name: str) -> Optional
         if scope and scope.ctes and name in scope.ctes:
             return scope.ctes[name]
     return None
+
+
+class SavedQueryVisitor(Resolver):
```

Comment:
This shouldn't inherit from `Resolver`, as that brings a lot of baggage and even more redundant steps and cloning of class trees. All visitors should inherit from `Visitor`, `TraversingVisitor` or `CloningVisitor`.